kernels/ZendEngine3/object.c

```diff
@@ -1194,7 +1194,7 @@ int zephir_create_instance_params(zval *return_value, const zval *class_name, co
 			zval *item;
 			int i = 0;
 
-			if (likely(param_count) <= 10) {
+			if (EXPECTED(param_count <= 10)) {
```

Comment:
@Jurigag no, your fix is fine, too. Your fix addresses the issue with the way how parameters are passed, my fix addresses the case when there are too many parameters and they corrupt the stack.

Your test case passes 11 parameters, the 11th one possibly overwrites `static_params` variable (it depends on the stack layout and how the compiler arranged local variables). But if you increase the number of parameters, they will eventually overwrite the stack pointer and crash the application.

Actually this is why the issue is compiler-specific: VC does not have __builtin_constant_p therefore the code works. clang could probably lay out the local variables differently, and the app does not crash. But gcc 4.8 does what it does, and the 11th parameter overwrites the stack pointer.